cmd/soroban-rpc/internal/config/options.go

```diff
@@ -272,68 +273,128 @@ func (cfg *Config) options() ConfigOptions {
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-events-queue-limit"),
 			Usage:        "Maximum number of outstanding GetEvents requests",
 			ConfigKey:    &cfg.RequestBacklogGetEventsQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-network-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-network-queue-limit"),
 			Usage:        "Maximum number of outstanding GetNetwork requests",
 			ConfigKey:    &cfg.RequestBacklogGetNetworkQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-latest-ledger-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-latest-ledger-queue-limit"),
 			Usage:        "Maximum number of outstanding GetLatestsLedger requests",
 			ConfigKey:    &cfg.RequestBacklogGetLatestLedgerQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-ledger-entries-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-ledger-entries-queue-limit"),
 			Usage:        "Maximum number of outstanding GetLedgerEntries requests",
 			ConfigKey:    &cfg.RequestBacklogGetLedgerEntriesQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding GetTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogGetTransactionQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-send-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-send-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding SendTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogSendTransactionQueueLimit,
 			DefaultValue: uint(500),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-simulate-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-simulate-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding SimulateTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogSimulateTransactionQueueLimit,
 			DefaultValue: uint(100),
 			Validate:     positive,
 		},
+		{
+			TomlKey:      strutils.KebabToConstantCase("request-execution-warning-threshold"),
+			Usage:        "The request execution warning threshold is the predetermined maximum duration of time that a request can take to be processed before a warning would be generated",
+			ConfigKey:    &cfg.RequestExecutionWarningThreshold,
+			DefaultValue: 5 * time.Second,
+		},
+		{
+			TomlKey:      strutils.KebabToConstantCase("max-request-execution-duration"),
```

Comment:
absorbing the last review and seeing similar number of new settings repeated for max duration, having a thought, wdyt about a de-duped config model down to 3 settings based on type/profile of request:

'default` is for all endpoints that have no special constraints, ie. they may read the local db and that's it 
`simulation` for simulate transaction
`submit tx` for submitting a transaction

each of the config types has a suffix for `-request-queue-limit` and `-execution-duration`, so the total permutations is 3 types x 2 suffixes = 6 configs

in this way, if new endpoints are added to rpc, don't need to change any configs if the new endpoint aligns with and can be governed by existing type thresholds, or worst case add a new config type for the new endpoint if needed.

this would remove the need to have the catch-all `max-request-execution-duration` as each type provides the notion of most effective threshold for it's grouping.

Reply:
I think that you've also identified the need to have different vastly different default for different limiters on different methods. The challenge that I see with grouping is that it would create a mapping that might represent a constraint when trying to adjust these values on a deployed platform.

Now, if we could argue that all/most of these endpoints have the exact same characteristics and timing, I would aside with your opinion. ( for instance, if we had 5 methods that are in-memory, immediate value returning ) 
At this time, we have quite a range of method implementations, and I honestly don't know what would be the usage pattern.

I would hope to get to the same level of flexibility, where we could "toggle" a single endpoint/method when a high-load scenario takes place.